[glm] Build failure

Package: glm:wasm32-emscripten -> 1.0.0

**Host Environment**

- Host: x64-windows
- Compiler: Clang 18.0.0
-    vcpkg-tool version: 2023-09-15-ac02a9f660977426b8ec6392919fbb1d51b10998
    vcpkg-scripts version: 3f806e04f 2024-02-07 (3 hours ago)

**To Reproduce**

`vcpkg install `
**Failure logs**

```
-- EMSDK PATH = C:/apps/emscripten/emsdk/
-- EMSCRIPTEN PATH = C:/apps/emscripten/emsdk/upstream/emscripten/
-- Using cached g-truc-glm-1.0.0.tar.gz.
-- Cleaning sources at C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/apps/vcpkg/vcpkg/downloads/g-truc-glm-1.0.0.tar.gz
-- Using source at C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean
-- Found external ninja('1.11.1').
-- Configuring wasm32-emscripten
-- Building wasm32-emscripten-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j9
    Working Directory: C:/apps/vcpkg/vcpkg/buildtrees/glm/wasm32-emscripten-dbg
    See logs for more information:
      C:\apps\vcpkg\vcpkg\buildtrees\glm\install-wasm32-emscripten-dbg-out.log

Call Stack (most recent call first):
  C:/dev/Graphics/code/build-web/vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  C:/dev/Graphics/code/build-web/vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/glm/portfile.cmake:20 (vcpkg_cmake_install)
  scripts/ports.cmake:170 (include)



```
<details><summary>C:\apps\vcpkg\vcpkg\buildtrees\glm\install-wasm32-emscripten-dbg-out.log</summary>

```
Change Dir: 'C:/apps/vcpkg/vcpkg/buildtrees/glm/wasm32-emscripten-dbg'

Run Build Command(s): C:/tools/ninja/bin/ninja.exe -v -v -j9 install
[1/3] C:\apps\emscripten\emsdk\upstream\emscripten\em++.bat -DGLM_FORCE_CXX17 -IC:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean -g -std=c++17 -Wno-c++98-compat -Wno-c++98-compat-pedantic -Werror -Weverything -MD -MT glm/CMakeFiles/glm.dir/detail/glm.cpp.o -MF glm\CMakeFiles\glm.dir\detail\glm.cpp.o.d -o glm/CMakeFiles/glm.dir/detail/glm.cpp.o -c C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean/glm/detail/glm.cpp
FAILED: glm/CMakeFiles/glm.dir/detail/glm.cpp.o 
C:\apps\emscripten\emsdk\upstream\emscripten\em++.bat -DGLM_FORCE_CXX17 -IC:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean -g -std=c++17 -Wno-c++98-compat -Wno-c++98-compat-pedantic -Werror -Weverything -MD -MT glm/CMakeFiles/glm.dir/detail/glm.cpp.o -MF glm\CMakeFiles\glm.dir\detail\glm.cpp.o.d -o glm/CMakeFiles/glm.dir/detail/glm.cpp.o -c C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean/glm/detail/glm.cpp
In file included from C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean/glm/detail/glm.cpp:7:
In file included from C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx/dual_quaternion.hpp:19:
In file included from C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\../glm.hpp:133:
In file included from C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\packing.hpp:173:
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:20:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   20 |                 u.in[1] = result[1];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:35:25: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   35 |                 return vec2(u.out[0], u.out[1]) * 1.5259021896696421759365224689097e-5f;
      |                                       ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:49:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   49 |                 u.in[1] = result[1];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:64:31: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   64 |                 return clamp(vec2(u.out[0], u.out[1]) * 3.0518509475997192297128208258309e-5f, -1.0f, 1.0f);
      |                                             ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:78:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   78 |                 u.in[1] = result[1];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:79:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   79 |                 u.in[2] = result[2];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:80:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   80 |                 u.in[3] = result[3];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:95:25: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   95 |                 return vec4(u.out[0], u.out[1], u.out[2], u.out[3]) * 0.0039215686274509803921568627451f;
      |                                       ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:95:35: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   95 |                 return vec4(u.out[0], u.out[1], u.out[2], u.out[3]) * 0.0039215686274509803921568627451f;
      |                                                 ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:95:45: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
   95 |                 return vec4(u.out[0], u.out[1], u.out[2], u.out[3]) * 0.0039215686274509803921568627451f;
      |                                                           ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:109:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  109 |                 u.in[1] = result[1];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:110:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  110 |                 u.in[2] = result[2];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:111:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  111 |                 u.in[3] = result[3];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:126:31: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  126 |                 return clamp(vec4(u.out[0], u.out[1], u.out[2], u.out[3]) * 0.0078740157480315f, -1.0f, 1.0f);
      |                                             ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:126:41: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  126 |                 return clamp(vec4(u.out[0], u.out[1], u.out[2], u.out[3]) * 0.0078740157480315f, -1.0f, 1.0f);
      |                                                       ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:126:51: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  126 |                 return clamp(vec4(u.out[0], u.out[1], u.out[2], u.out[3]) * 0.0078740157480315f, -1.0f, 1.0f);
      |                                                                 ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:138:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  138 |                 u.in[1] = v[1];
      |                 ^~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:153:26: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  153 |                 return uvec2(u.out[0], u.out[1]);
      |                                        ^~~~~
C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean\glm/gtx\..\detail/func_packing.inl:165:3: error: unsafe buffer access [-Werror,-Wunsafe-buffer-usage]
  165 |                 u.in[1] = detail::toFloat16(v.y);
      |                 ^~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
em++: error: 'C:/apps/emscripten/emsdk/upstream/bin\clang++.exe -target wasm32-unknown-emscripten -fignore-exceptions -fvisibility=default -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr --sysroot=C:\apps\emscripten\emsdk\upstream\emscripten\cache\sysroot -DEMSCRIPTEN -Xclang -iwithsysroot/include\fakesdl -Xclang -iwithsysroot/include\compat -DGLM_FORCE_CXX17 -IC:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean -g3 -std=c++17 -Wno-c++98-compat -Wno-c++98-compat-pedantic -Werror -Weverything -MD -MT glm/CMakeFiles/glm.dir/detail/glm.cpp.o -MF glm\CMakeFiles\glm.dir\detail\glm.cpp.o.d -c C:/apps/vcpkg/vcpkg/buildtrees/glm/src/1.0.0-754ce17bcc.clean/glm/detail/glm.cpp -o glm/CMakeFiles/glm.dir/detail/glm.cpp.o' failed (returned 1)

ninja: build stopped: subcommand failed.
```
</details>**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "dependencies": [
    "glm",
    "gtest",
    "entt",
    "assimp",
    "bullet3",
    "lodepng",
    "stb",
    "freetype",
    {
      "name": "opengl",
      "platform": "!emscripten"
    },
    {
      "name": "openal-soft",
      "platform": "!emscripten"
    },
    {
      "name": "glfw3",
      "platform": "!emscripten"
    },
    {
      "name": "glew",
      "platform": "!emscripten"
    }
  ]
}

```
</details>

--- a/ports/glm/disable-werror.patch
+++ b/ports/glm/disable-werror.patch
@@ -1,0 +1,119 @@
+From 7b53739128fe99b170011ab8c2733b7f56d93116 Mon Sep 17 00:00:00 2001
+From: Christophe <christophe@lunarg.com>
+Date: Tue, 6 Feb 2024 16:06:52 +0100
+Subject: [PATCH] Only enable compiler warnings and warnings as error on unit
+ tests
+
+---
+ CMakeLists.txt      | 43 -------------------------------------------
+ readme.md           |  3 +++
+ test/CMakeLists.txt | 41 +++++++++++++++++++++++++++++++++++++++++
+ 3 files changed, 44 insertions(+), 43 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 217b76bf3..034177220 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -244,49 +244,6 @@ elseif(GLM_ENABLE_SIMD_SSE2)
+ 	message(STATUS "GLM: SSE2 instruction set")
+ endif()
+ 
+-# Compiler and default options
+-
+-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+-	if(NOT GLM_QUIET)
+-		message("GLM: Clang - ${CMAKE_CXX_COMPILER_ID} compiler")
+-	endif()
+-
+-	if(NOT GLM_DISABLE_AUTO_DETECTION)
+-		add_compile_options(-Werror -Weverything)
+-	endif()
+-#	add_compile_options(-Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-c++11-long-long -Wno-padded -Wno-gnu-anonymous-struct -Wno-nested-anon-types)
+-#	add_compile_options(-Wno-undefined-reinterpret-cast -Wno-sign-conversion -Wno-unused-variable -Wno-missing-prototypes -Wno-unreachable-code -Wno-missing-variable-declarations -Wno-sign-compare -Wno-global-constructors -Wno-unused-macros -Wno-format-nonliteral -Wno-float-equal)
+-
+-elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+-	if(NOT GLM_QUIET)
+-		message("GLM: GCC - ${CMAKE_CXX_COMPILER_ID} compiler")
+-	endif()
+-
+-	if(NOT GLM_DISABLE_AUTO_DETECTION)
+-		add_compile_options(-Werror)
+-#		add_compile_options(-Wpedantic)
+-#		add_compile_options(-Wall)
+-#		add_compile_options(-Wextra)
+-	endif()
+-	add_compile_options(-O2)
+-	#add_compile_options(-Wno-long-long)
+-elseif(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+-	if(NOT GLM_QUIET)
+-		message("GLM: Intel - ${CMAKE_CXX_COMPILER_ID} compiler")
+-	endif()
+-
+-elseif(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+-	if(NOT GLM_QUIET)
+-		message("GLM: Visual C++ - ${CMAKE_CXX_COMPILER_ID} compiler")
+-	endif()
+-
+-	if(NOT GLM_DISABLE_AUTO_DETECTION)
+-		add_compile_options(/W4 /WX)
+-	endif()
+-#	add_compile_options(/wd4309 /wd4324 /wd4389 /wd4127 /wd4267 /wd4146 /wd4201 /wd4464 /wd4514 /wd4701 /wd4820 /wd4365)
+-	add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+-endif()
+-
+ add_subdirectory(glm)
+ 
+ if (GLM_BUILD_TESTS)
+
+diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
+index abfa33451..9272c591d 100644
+--- a/test/CMakeLists.txt
++++ b/test/CMakeLists.txt
+@@ -3,6 +3,47 @@ option(GLM_QUIET "No CMake Message" OFF)
+ option(GLM_TEST_ENABLE "Build unit tests" ON)
+ option(GLM_PERF_TEST_ENABLE "Build perf tests" OFF)
+ 
++# Compiler and default options
++
++if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
++	if(NOT GLM_QUIET)
++		message("GLM: Clang - ${CMAKE_CXX_COMPILER_ID} compiler")
++	endif()
++
++	if(NOT GLM_DISABLE_AUTO_DETECTION)
++		add_compile_options(-Werror -Weverything)
++	endif()
++
++elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
++	if(NOT GLM_QUIET)
++		message("GLM: GCC - ${CMAKE_CXX_COMPILER_ID} compiler")
++	endif()
++
++	if(NOT GLM_DISABLE_AUTO_DETECTION)
++		add_compile_options(-Werror)
++#		add_compile_options(-Wpedantic)
++#		add_compile_options(-Wall)
++#		add_compile_options(-Wextra)
++	endif()
++	add_compile_options(-O2)
++	#add_compile_options(-Wno-long-long)
++elseif(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
++	if(NOT GLM_QUIET)
++		message("GLM: Intel - ${CMAKE_CXX_COMPILER_ID} compiler")
++	endif()
++
++elseif(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
++	if(NOT GLM_QUIET)
++		message("GLM: Visual C++ - ${CMAKE_CXX_COMPILER_ID} compiler")
++	endif()
++
++	if(NOT GLM_DISABLE_AUTO_DETECTION)
++		add_compile_options(/W4 /WX)
++	endif()
++#	add_compile_options(/wd4309 /wd4324 /wd4389 /wd4127 /wd4267 /wd4146 /wd4201 /wd4464 /wd4514 /wd4701 /wd4820 /wd4365)
++	add_definitions(-D_CRT_SECURE_NO_WARNINGS)
++endif()
++
+ function(glmCreateTestGTC NAME)
+ 	set(SAMPLE_NAME test-${NAME})
+ 	add_executable(${SAMPLE_NAME} ${NAME}.cpp)

--- a/ports/glm/portfile.cmake
+++ b/ports/glm/portfile.cmake
@@ -6,6 +6,8 @@ vcpkg_from_github(
     REF "${VERSION}"
     SHA512 62e22002a6369a54e1f0ee2885a65f2780af7d2a446573e5387b81518f5dc7e8076053837cb99ae850a0166ce8b0f077bed009e8986d9884d01c456ce467553f
     HEAD_REF master
+    PATCHES
+        "disable-werror.patch" # https://github.com/g-truc/glm/commit/7b53739128fe99b170011ab8c2733b7f56d93116
 )
 
 vcpkg_cmake_configure(

--- a/ports/glm/vcpkg.json
+++ b/ports/glm/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "glm",
   "version": "1.0.0",
+  "port-version": 1,
   "description": "OpenGL Mathematics (GLM)",
   "homepage": "https://glm.g-truc.net",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3038,7 +3038,7 @@
     },
     "glm": {
       "baseline": "1.0.0",
-      "port-version": 0
+      "port-version": 1
     },
     "globjects": {
       "baseline": "1.1.0",

--- a/versions/g-/glm.json
+++ b/versions/g-/glm.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "1a0115a7d1f3c29d3ebfa85ed25914d077dbc6a3",
+      "version": "1.0.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "ed558e712eafccb124025c665c39f21998c71438",
       "version": "1.0.0",
       "port-version": 0

